docs/examples/object_comparator_t.cpp

```diff
@@ -0,0 +1,11 @@
+#include <iostream>
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+
+int main()
+{
+    std::cout << std::boolalpha
+              << "one < two : " << json::object_comparator_t{}("one", "two") << "\n"
+              << "three < four : " << json::object_comparator_t{}("three", "four") << std::endl;
```

Comment:
`object_comparator_t` is not guaranteed to be `std::less` anymore. Could also be  `std::equal_to` or possibly something else depending on `object_t::key_compare`.

Reply:
What about

```cpp
std::cout << std::boolalpha
          << "json::object_comparator_t{}(\"one\", \"two\") = " << json::object_comparator_t{}("one", "two") << "\n"
          << "json::object_comparator_t{}(\"three\", \"four\") = : " << json::object_comparator_t{}("three", "four") << std::endl;
```

?

Reply:
Sure. (Without the erroneous ':' in `\"four\") = : "`.)

Reply:
Done. (The code above never saw an IDE ;-))